src/main/java/seedu/address/model/restaurant/Restaurant.java

```diff
@@ -107,6 +111,7 @@ public Restaurant(Restaurant restaurant, Category category) {
         this.categories = category;
         this.weblink = restaurant.weblink;
         this.openingHours = restaurant.openingHours;
+        this.summary = new Summary(restaurant.getReviews());
     }
 
     public Name getName() {
```

Comment:
Instead of using getters, would be better to create a method. For example, it is possible to pass the UI components into a method to set the name values and summary data.